### DISCUSSION TITLE
A little question about demo

### DISCUSSION THREAD

**Discussion by possible7** [lululxvi/deepxde#1374] [Q&A]
Dear @lululxvi,
Thanks for giving the demo for us. It's very useful for my research. However, when I run the demo "Volterra_IDE.py", the code can't work. If I change the model compile (L-BFGS to adam), it can get the answer. I want to ask is there any way to solve it？
In addition, I want to know which equation is solved by the demo" ide.py".

Happy to hear about your opinion and any suggestions!

Best,
Zan Yu

**Comment by lululxvi** [comment]
> However, when I run the demo "Volterra_IDE.py", the code can't work. If I change the model compile (L-BFGS to adam), it can get the answer. I want to ask is there any way to solve it？

What is the error?

**Reply by possible7** [reply]
The error is as follows：
---------------------------------------------------------------------------
InvalidArgumentError                      Traceback (most recent call last)
File [d:\anaconda3\envs\test\lib\site-packages\tensorflow\python\client\session.py:1378](file:///D:/anaconda3/envs/test/lib/site-packages/tensorflow/python/client/session.py:1378), in BaseSession._do_call(self, fn, *args)
   1377 try:
-> 1378   return fn(*args)
   1379 except errors.OpError as e:

File [d:\anaconda3\envs\test\lib\site-packages\tensorflow\python\client\session.py:1361](file:///D:/anaconda3/envs/test/lib/site-packages/tensorflow/python/client/session.py:1361), in BaseSession._do_run.._run_fn(feed_dict, fetch_list, target_list, options, run_metadata)
   1360 self._extend_graph()
-> 1361 return self._call_tf_sessionrun(options, feed_dict, fetch_list,
   1362                                 target_list, run_metadata)

File [d:\anaconda3\envs\test\lib\site-packages\tensorflow\python\client\session.py:1454](file:///D:/anaconda3/envs/test/lib/site-packages/tensorflow/python/client/session.py:1454), in BaseSession._call_tf_sessionrun(self, options, feed_dict, fetch_list, target_list, run_metadata)
   1452 def _call_tf_sessionrun(self, options, feed_dict, fetch_list, target_list,
   1453                         run_metadata):
-> 1454   return tf_session.TF_SessionRun_wrapper(self._session, options, feed_dict,
   1455                                           fetch_list, target_list,
   1456                                           run_metadata)

InvalidArgumentError: Matrix size-incompatible: In[0]: [12,252], In[1]: [253,1]
	 [[{{node MatMul_1}}]]

During handling of the above exception, another exception occurred:

InvalidArgumentError                      Traceback (most recent call last)
...
  File "d:\anaconda3\envs\test\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 795, in _apply_op_helper
    op = g._create_op_internal(op_type_name, inputs, dtypes=None,
  File "d:\anaconda3\envs\test\lib\site-packages\tensorflow\python\framework\ops.py", line 3814, in _create_op_internal
    ret = Operation(
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?3d55d69f-021a-4f2b-a020-14672d94db06) or open in a [text editor](command:workbench.action.openLargeOutput?3d55d69f-021a-4f2b-a020-14672d94db06). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...

**Reply by possible7** [reply]
In addition, when I change the version of deepxde to 1.7.0, I can get similar results to the paper.

**Reply by lululxvi** [reply]
The error information is not complete, but it seems a problem of Tensorflow.

**Reply by possible7** [reply]
Is there any solution, or is it necessary to change a backend?

**Reply by lululxvi** [reply]
I don't know the solution given the information you give. But you can try other backends.
